proxy/src/telemetry/metrics/latency.rs

```diff
@@ -62,10 +63,11 @@ pub struct Histogram {
     /// Array of buckets in which to count latencies.
     ///
     /// The upper bound of a given bucket `i` is given in `BUCKET_BOUNDS[i]`.
-    buckets: [u32; NUM_BUCKETS],
+    // TODO: consider using `prometheus::Counter` rather than `Wrapping<u32>`?
+    buckets: [Wrapping<u32>; NUM_BUCKETS],
 
     /// The total sum of all observed latency values.
-    pub sum: u64,
+    pub sum: Wrapping<u64>,
```

Comment:
Shouldn't this be `prometheus::Counter` too?

Reply:
Hmm. The `sum` is incremented by the value of an observation, not by 1. My intention in having `prometheus::Counter` expose a public `.incr()` method but no `Add`/`AddAssign` impls was to constrain the possible operations on a counter. Although I suppose `prometheus::Counter` could also expose an `incr_by()` (or `AddAssign` ) impl as well and still constrain the allowed operations to addition only (i.e. enforce that it's monotonically increasing except in the event of overflows)...